hazelcast/src/main/java/com/hazelcast/map/impl/operation/GetOperation.java

```diff
@@ -38,7 +39,10 @@ public GetOperation(String name, Data dataKey) {
 
     @Override
     public void run() {
-        result = mapServiceContext.toData(recordStore.get(dataKey, false, getCallerAddress()));
+        Object currentValue = recordStore.get(dataKey, false, getCallerAddress());
+        result = (currentValue instanceof Immutable)
```

Comment:
Having immutable only makes sense if your application accesses it over its local embedded node, otherwise serialization will happen if you reach data over client or over remote nodes to the application. Is this what you want? Or maybe you can consider using near cache for this case.

Reply:
Think about use cases with Jet (access happening node locally from with a Jet pipeline)